File: desh/ege2026kp/12solve/12-349.cpp
// Автор: М. Нехорошева

#include<iostream>
using namespace std;
int main(){
	for(int n = 0; n < 10000; ++n) {
		string s = string(1,'7') + string(n + 1,'1') + string(n + 2,'2') + string(n + 3,'3');
		for(; s.find("71") != -1 || s.find("72") != -1 || s.find("73") != -1 ;){
			if( s.find("71") != -1 )
				s.replace(s.find("71"),2,"227");
			if( s.find("72") != -1 )
				s.replace(s.find("72"),2,"37");
			if( s.find("73") != -1 )
				s.replace(s.find("73"),2,"17");		
		}
		int sum = 0;
		sum = (n+2)*3 + (n+1)*4 + (n+3) + 7;
		if(sum == 9 * n) {
			cout<<n<<endl;
			break;
		}
	}
	
}

File: desh/ege2026kp/12solve/12-349.py

for n in range(1, 1000):
  s = '7' + '1'*(n+1) + '2'*(n+2) + '3'*(n+3)
  while '71' in s or '72' in s or '73' in s:
    s = s.replace( '71', '227', 1)
    s = s.replace( '72', '37', 1)
    s = s.replace( '73', '17', 1)
  sumDig = sum( map(int, s) )
  if sumDig == 9*n:
    print(n)
    break
